rt/frontend/simple/src/main/java/org/apache/cxf/frontend/WSDLGetUtils.java

```diff
@@ -566,9 +566,6 @@ private void processSchemaReference(SchemaReference schemaReference,
                         
                         doneSchemas.put(decodedStart, schemaReference);
                         doneSchemas.put(schemaReference.getReferencedSchema().getDocumentBaseURI(), schemaReference);
-                        if (!doneSchemas.containsKey(decodedOrigLocation)) {
```

Comment:
@reta @dkulp Any news on this? Do you want any modification on the PR?

Reply:
@rmartinc my apologies for the delay, I would definitely prefer for @dkulp taking a look (I am missing context for initial implementation), but I believe we could make the change a bit safer (please feel free to suggest better way comparing schema references):

```suggestion
if (!doneSchemas.containsKey(decodedOrigLocation)) {
    doneSchemas.put(decodedOrigLocation, schemaReference);
} else { 
    final SchemaReference ref = doneSchemas.get(decodedOrigLocation);
    // Conflict detected: different schemas under same decodedOrigLocation
    if (!ref.getSchemaLocationURI().equals(schemaReference.getSchemaLocationURI())) {
        doneSchemas.remove(decodedOrigLocation);
    }
}
```

Basically, the idea is - stick to previous behavior till the conflict is detected, wdyt @rmartinc ? thank you!

Reply:
OK @reta, no problem, we can wait a bit more for @dkulp.

I understand your idea, but it would not be in the second if, the change would go in the first one, something like this:

```diff
--- a/rt/frontend/simple/src/main/java/org/apache/cxf/frontend/WSDLGetUtils.java
+++ b/rt/frontend/simple/src/main/java/org/apache/cxf/frontend/WSDLGetUtils.java
@@ -552,7 +552,9 @@ public class WSDLGetUtils {
                                 start), e);
             }
 
-            if (!doneSchemas.containsKey(decodedStart)) {
+            final SchemaReference doneSchema = doneSchemas.get(decodedStart);
+            if (doneSchema == null || !doneSchema.getSchemaLocationURI().equals(
+                    schemaReference.getSchemaLocationURI())) {
                 String resolvedSchemaLocation = resolveWithCatalogs(catalogs, start, base);
                 if (resolvedSchemaLocation == null) {
                     resolvedSchemaLocation =
```
Note that the problem is that the incorrect schema is added first, so we cannot do anything at that point. The idea is adding the second one if not exists or if it is different in URI (it's another schema).

I have tested that solution and it works good for my case. No issues detected in the TS either.

Reply:
Thanks @rmartinc , may I ask you please to update the pull request with the change you suggest? It is a little bit difficult (to me) to understand the it, thank you.

Reply:
Done! The idea is the same one you presented before but in the other if.

Previously the schema was just discarded if it was already in the `doneSchemas` map. But now it is checked if it's the same too (same uri). If it's different then it's managed and the `decodedStart` is reset with the new schema. That way the previous one added is overwritten.

IIUIC the problem is that the `decodedStart` is different between the different files but it can collide with any other `decodedOrigLocation` (if they happen to use the same include, but in different directory, as in the example). Current implementation works if the `decodedStart` was inserted in the map before the `decodedOrigLocation`, but not the other way around. With this change we are always adding the `decodedStart`, no matter if it was assigned before by a `decodedOrigLocation` of another schema.